src/Workspaces/MSBuildTest/Utilities/VisualStudioMSBuildInstalled.cs

```diff
@@ -3,46 +3,48 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Linq;
 using Microsoft.Build.Locator;
 using Roslyn.Test.Utilities;
 
 namespace Microsoft.CodeAnalysis.MSBuild.UnitTests
 {
     internal class VisualStudioMSBuildInstalled : ExecutionCondition
     {
+        private readonly Version _minimumVersion;
+
 #if NET472_OR_GREATER
+
         private static readonly VisualStudioInstance? s_instance;
-        private readonly Version _minimumVersion;
 
         static VisualStudioMSBuildInstalled()
         {
-            var installedVisualStudios = MSBuildLocator.QueryVisualStudioInstances().ToArray();
-            foreach (var visualStudioInstall in installedVisualStudios)
+            var latestInstalledInstance = (VisualStudioInstance?)null;
+            foreach (var visualStudioInstance in MSBuildLocator.QueryVisualStudioInstances())
             {
-                if (visualStudioInstall.Version.Major == 17 &&
-                    visualStudioInstall.Version.Minor == 0)
+                if (latestInstalledInstance == null || visualStudioInstance.Version > latestInstalledInstance.Version)
```

Comment:
do we still want a min for the oldest supported version for any reason?

Reply:
The oldest supported version gets filtered out in the other checks later in the file, since the minimum version can vary test to test.

Reply:
The min version check occurs in the test condition's `ShouldSkip` implementation.